pkg/manifests/manifests.go

```diff
@@ -24,15 +25,62 @@ import (
 	"github.com/Azure/aks-app-routing-operator/pkg/util"
 )
 
-const IngressClass = "webapprouting.kubernetes.azure.com"
+const (
+	IngressClass                = "webapprouting.kubernetes.azure.com"
+	appsV1ApiVersion            = "apps/v1"
+	deploymentKind              = "Deployment"
+	nsKind                      = "Namespace"
+	version                     = "v1"
+	ingressClassKind            = "IngressClass"
+	networkingApiVersion        = "networking.k8s.io/v1"
+	nginxIngressClassName       = "k8s.io/ingress-nginx"
+	serviceAccountKind          = "ServiceAccount"
+	clusterRoleKind             = "ClusterRole"
+	rbacApiVersion              = "rbac.authorization.k8s.io/v1"
+	clusterRoleBindingKind      = "ClusterRoleBinding"
+	serviceKind                 = "Service"
+	osmAnnotationKey            = "openservicemesh.io/sidecar-injection"
+	nginxIngressControllerImage = "/oss/kubernetes/ingress/nginx-ingress-controller:v1.2.1"
+	nginxCpuRequest             = "500m"
+	nginxMemoryRequest          = "127Mi"
+	nginxCpuLimit               = "1500m"
+	nginxMemoryLimit            = "512Mi"
+	hpaKind                     = "HorizontalPodAutoscaler"
+	autoscalingApiVersion       = "autoscaling/v1"
+	appKeySelector              = "app"
+	externalDnsDeploymentName   = "controller"
+	externalDnsImage            = "/oss/kubernetes/external-dns:v0.11.0.2"
+	externalDnsCpuRequest       = "100m"
+	externalDnsMemoryRequest    = "250Mi"
+	externalDnsCpuLimit         = "100m"
+	externalDnsMemoryLimit      = "250Mi"
+	externalDnsVolumeName       = "azure-config"
+)
 
 var (
 	IngressControllerName = "nginx"
-	IngressPodLabels      = map[string]string{"app": IngressControllerName}
-
-	externalDNSName = "external-dns"
+	IngressPodLabels      = map[string]string{appKeySelector: IngressControllerName}
+
+	externalDNSName         = "external-dns"
+	azurePrivateDNSProvider = "azure-private-dns"
+	ingressSource           = "ingress"
+
+	topLevelLabels             = map[string]string{"app.kubernetes.io/managed-by": "aks-app-routing-operator"}
+	nginxIngressControllerArgs = []string{
```

Comment:
this is another example of something that shouldn't be moved to the top. This only has context inside of the nginxIngressControllerDeployment. Moving it here provides no benefit and just makes it way harder to read the code

Reply:
makes sense - esp given that a good chunk of this will be ripped out anyway, I'll just reintroduce it as regular values to minimize diff